sequencer/src/catchup.rs

```diff
@@ -235,17 +241,36 @@ impl<T> StateCatchup for SqlStateCatchup<T>
 where
     T: CatchupDataSource + std::fmt::Debug + Send + Sync,
 {
+    // TODO: add a test for the account proof validation
+    // issue # 2102 (https://github.com/EspressoSystems/espresso-sequencer/issues/2102)
     #[tracing::instrument(skip(self))]
     async fn try_fetch_account(
         &self,
         block_height: u64,
         view: ViewNumber,
-        _fee_merkle_tree_root: FeeMerkleCommitment,
+        fee_merkle_tree_root: FeeMerkleCommitment,
         account: FeeAccount,
     ) -> anyhow::Result<AccountQueryData> {
-        self.db
+        let res = self
+            .db
             .get_account(block_height, view, account.into())
-            .await
+            .await?;
+
+        if res.proof.account != account.into() {
+            panic!(
+                "Critical error: Mismatched fee account proof: expected {:?}, got {:?}.
+                This may indicate a compromised database",
+                account, res.proof.account
+            );
+        }
+
+        match res.proof.verify(&fee_merkle_tree_root) {
```

Comment:
I think `verify()` method should not return `anyhow` and  `FeeAccountProof` should have a dedicated error. This way we can clean up caller logic and also write better tests.

Reply:
https://github.com/EspressoSystems/espresso-sequencer/issues/2146